public/app/features/templating/template_srv.ts

```diff
@@ -347,6 +349,21 @@ export class TemplateSrv implements BaseTemplateSrv {
     });
   }
 
+  getVariablesMapInTemplate(target: string, scopedVars: ScopedVars): Record<string, VarValue> {
+    const regexp = new RegExp(this.regex);
+    const values: Record<string, VarValue> = {};
+
+    target.replace(regexp, (match, var1, var2, fmt2, var3, fieldPath) => {
```

Comment:
I feel like we want to keep this target.replace usage in template_srv just to ensure it is always consistent with the replace function in this file. Usage of it though should go outside this function. What do you think?

Reply:
Yeah, to me it feels that `this.regex` is internal to the `template_srv` (so is the `replace` that is using it directly). It probably doesn't make sense to use it outside.